manage_breast_screening/mammograms/forms/other_procedure_history_form.py

```diff
@@ -0,0 +1,112 @@
+from django.forms.widgets import Textarea
+
+from manage_breast_screening.core.services.auditor import Auditor
+from manage_breast_screening.nhsuk_forms.fields import CharField, ChoiceField
+from manage_breast_screening.nhsuk_forms.fields.integer_field import YearField
+from manage_breast_screening.nhsuk_forms.forms import FormWithConditionalFields
+from manage_breast_screening.participants.models.other_procedure_history_item import (
+    OtherProcedureHistoryItem,
+)
+
+
+class OtherProcedureHistoryForm(FormWithConditionalFields):
+    PROCEDURE_DETAIL_FIELDS = {
+        OtherProcedureHistoryItem.Procedure.BREAST_REDUCTION: "breast_reduction_details",
+        OtherProcedureHistoryItem.Procedure.BREAST_SYMMETRISATION: "breast_symmetrisation_details",
+        OtherProcedureHistoryItem.Procedure.NIPPLE_CORRECTION: "nipple_correction_details",
+        OtherProcedureHistoryItem.Procedure.OTHER: "other_details",
+    }
+
+    def __init__(self, *args, participant, **kwargs):
+        self.instance = kwargs.pop("instance", None)
+
+        if self.instance:
+            kwargs["initial"] = self.initial_values(self.instance)
+
+        super().__init__(*args, **kwargs)
+
+        self.fields["procedure"] = ChoiceField(
+            choices=OtherProcedureHistoryItem.Procedure,
+            label=f"What procedure has {participant.first_name} had?",
+            error_messages={"required": "Select the procedure"},
+        )
+
+        for procedure, detail_field in self.PROCEDURE_DETAIL_FIELDS.items():
+            self.fields[detail_field] = CharField(
+                required=False,
+                label="Provide details",
+                error_messages={"required": "Provide details of the procedure"},
```

Comment:
@edwardhorsford Do we need to describe the field here?

Whilst the conditional reveal only shows one field at once, there are three in the HTML

* **Label:** Provide details [of the nipple correction procedue]
* **Error:** ~Provide~ Enter details of the [nipple correction] procedue

Reply:
I think Danny has a ticket to review all the error messages so after that we'll have another dev ticket to update the all the existing forms. Maybe as part of that we can sort out the labels as well. I think we've got issues like this in a couple of other places.